- Credit
    - http://www.cyclismo.org/tutorial/R/confidence.html

In [1]:
import pandas as pd, numpy as np
%matplotlib inline
%pylab inline
import seaborn  as sns 
import matplotlib.pyplot as plt
plt.style.use('ggplot')

Populating the interactive namespace from numpy and matplotlib


### 1-1) Calculating a Confidence Interval From a Normal Distribution

In [4]:
from scipy.stats import norm

In [9]:
# quantile function in python
# R 
# https://stat.ethz.ch/R-manual/R-devel/library/stats/html/Normal.html
# PYTHON 
# https://stackoverflow.com/questions/24695174/python-equivalent-of-qnorm-qf-and-qchi2-of-r
# https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

# qnorm(0.975) IN R 
norm.ppf(.975)

1.959963984540054

In [14]:
# assume 
# sample mean = 5,
# standard deviation = 2
# sample size = 20. 
# use a 95% confidence level 

a = 5 
s = -2 
n = 20
error = norm.ppf(.975)*s/np.sqrt(20)
left = a-error
right = a+error

print ('left: {}'.format(left))
print ('right: {}'.format(right))



left: 5.876522540576581
right: 4.123477459423419


`The mean of predicting values are within confidence interval between 4.12 and 5.88 with 95% confidence interval, data is  normally distributed and samples are independent`


### 1-2)  Calculating a Confidence Interval From a t Distribution

In [22]:
from scipy.stats import t

In [25]:
# quantile function in the The Student t Distribution
# R 
# https://stat.ethz.ch/R-manual/R-devel/library/stats/html/TDist.html
# PYTHON
# https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.t.html
# https://stackoverflow.com/questions/19339305/python-function-to-get-the-t-statistic

# qt(0.975,df=n-1) IN R 
t.ppf(.975, df=n-1)

2.093024054408263

In [21]:
# assume 
# sample mean = 5,
# standard deviation = 2
# sample size = 20. 
# use a 95% confidence level 

a = 5 
s = -2 
n = 20
error = t.ppf(.975, df=n-1)*s/np.sqrt(20)
left = a-error
right = a+error

print ('left: {}'.format(left))
print ('right: {}'.format(right))



left: 5.936028812839819
right: 4.063971187160181


`The true mean has a probability of 95% of being in the interval between 4.06 and 5.94 assuming that the original random variable is normally distributed, and the samples are independent.`

### 1-3) Calculating a Confidence Interval From a DataFrame

In [26]:
df = pd.read_csv('http://www.cyclismo.org/tutorial/R/_static/w1.dat')

In [28]:
df.head(3)

vals
0  0.43
1  0.40
2  0.45

In [30]:
df.describe().T

count   mean       std   min   25%   50%     75%   max
vals   54.0  0.765  0.378122  0.13  0.48  0.72  1.0075  1.76

In [42]:
error = t.ppf(.975, df=len(df)-1)*std(df)/np.sqrt(len(df))

print ('error : {}'.format(error))

# R 
# error <- qt(0.975,df=length(w1$vals)-1)*std(df)/sqrt(length(w1$vals))


error : vals    0.102247
dtype: float64


In [44]:
left = mean(df)-error
right = mean(df)+error

print ('left: {}'.format(left))
print ('right: {}'.format(right))

left: vals    0.662753
dtype: float64
right: vals    0.867247
dtype: float64


`There is a 95% probability that the true mean is between 0.66 and 0.87 assuming that the original random variable is normally distributed, and the samples are independent.`